compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraph.skiko.kt

```diff
@@ -496,12 +496,16 @@ internal class SkiaParagraph(
         val leftX = rects?.firstOrNull()?.rect?.left ?: expectedLine.left.toFloat()
         val rightX = rects?.lastOrNull()?.rect?.right ?: expectedLine.right.toFloat()
 
-        if (leftX == rightX) {
+        // Skia might've returned the glyphPosition that belongs to a different line :|
+        val lineMatch = glyphPosition >= expectedLine.startIndex && glyphPosition <= expectedLine.endIndex
```

Comment:
Is `endIndex` inclusive or exclusive? Shouldn't it be `glyphPosition < expectedLine.endIndex`?

Reply:
> > Accidentally caught [a native crash](https://github.com/JetBrains/androidx/files/9620804/hs_err_pid14324.log) on this example (without your fix, on [11a65f9](https://github.com/JetBrains/androidx/commit/11a65f96f6e82ef3fbe63f217e58b09546a39ba1))
> 
> Reproduce it often now. But there is no pattern:
> 
>  java_7hgiKJ0NtF.mp4

It seems there is a pattern:

Crashes when dragging the cursor in the red area
<img width="791" alt="Screenshot 2022-09-22 at 11 35 06" src="https://user-images.githubusercontent.com/7372778/191712981-34d2533e-704a-476b-938f-25fdd64d9a20.png">

Reply:
The crash was fixed in skiko 0.7.34
The issue of getOffsetForPosition returning 0 on an empty line doesn't reproduce with skiko 0.7.34 too.

So for now, to avoid making the code even less clear, I'd just rely on skiko 0.7.34 and not merge this PR (with `val lineMatch = ...` check).